### PR TITLE
Autogenerated CV overview page

### PR DESCRIPTION
**Pull request by DunklesArchipel**
This adds an overview page for the digitized CVs.

image

There are some tweaks to this in #159.

Dependencies:
- [x] https://github.com/echemdb/echemdb/pull/20
- [x] https://github.com/echemdb/echemdb/pull/38

### CODE REVIEW

**Comment by DunklesArchipel** [comment]
@saraedum @linuxrider 
I implemented the above idea. Since some of the identifiers are rather long, the tables become rather broad and one has to scroll.

Maybe we shoudl remove it and instead have a material entry such as `Pt(111)` which becomes a link.
The Problem ist, that a lot of entries will look the same. For example there will be a plethora of entries named

| material | components | Ref |
| -- | -- | -- |
| Pt(111) | 0.5 M H2SO4 | REF1 |
| Pt(111) | 0.5 M H2SO4 | REF1 |
| Pt(111) | 0.5 M H2SO4 | REF2 |

where the first two REF1 entries simply point to different figures.

So maybe REF could be `Author et al. (YEAR), Fig 1a`

A `scan rate` could be a nice addition.

image

File: website/generator/__main__.py [hunk 1]
```diff
@@ -42,18 +42,38 @@ def main():
...
+                        "pages/cv.md",
+                        database=database
+                    )
+                )
+        for material in ['Ru', 'Cu', 'Pt']: #database.materials()
+            filtered_db = database.filter(lambda entry: entry.system.electrodes.working_electrode.material == material)
+            markdown.write(
+                render(
+                    "components/cv_table.md",
+                    database=filtered_db, material=material
```
Comment (DunklesArchipel) [inline website/generator/__main__.py:74]: This should be included in CV.md

**Comment by DunklesArchipel** [comment]
The new design looks as follows:
image

**Comment by DunklesArchipel** [comment]
I would be curios how we can render such names properly:
image

File: templates/pages/cv.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Cyclic Voltammograms
+
+Select a graph for more details.
+
+<!--
```
Comment (saraedum) [inline templates/pages/cv.md:13]: What's the plan with this comment?

I think it should be removed (and turned into an issue.)

File: website/generator/__main__.py [hunk 3]
```diff
@@ -42,17 +42,24 @@ def main():
...
             markdown.write(
                 render(
                     "pages/cv_entry.md",
-                    database=website.generator.database.cv,
+                    database=database,
                     entry=entry,
                 )
             )
+    # Create an overview page with tabulated and linked entries.
+    with mkdocs_gen_files.open(os.path.join("cv", "index.md"), "w") as markdown:
```
Comment (saraedum) [inline website/generator/__main__.py:60]: Why do we want to concatenate two files here? It's easier to just render `cv_overview_table.md` inside `cv.md`.

File: templates/components/cv_overview_table.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+| Entry | Substrate | Electrolyte | year | Reference |
```
Comment (saraedum) [inline templates/components/cv_overview_table.md:1]: @DunklesArchipel I changed this table to be more reusable. It now expects a database as input and renders a table, i.e., there is no reference to materials anymore.

**Comment by saraedum** [comment]
@DunklesArchipel please have another look here. It worked for me locally. We need to update the echemdb-feedstock for this to actually work in the CI.

File: templates/pages/cv.md (hunk 2 above)
Comment (saraedum) [reply templates/pages/cv.md]: This is now https://github.com/echemdb/website/issues/160.

**Review by saraedum — APPROVED** [review]

File: templates/pages/cv.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Cyclic Voltammograms
+
+Select a graph for more details.
```
Comment (saraedum) [inline templates/pages/cv.md:3]:
```suggestion
Click a graph for more details.
```

**Comment by saraedum** [comment]
@DunklesArchipel feel free to merge this if you agree with my changes.

**Comment by DunklesArchipel** [comment]
Looks great! thanks a lot.

**Pull request merged**